# Downstream analysis of WGS assembly

## Table of Contents ##

1. [Detecct genes of interest of *Bacillus cereus* ](#btyper)([**Btyper3**](https://github.com/lmc297/BTyper3))
2. [Identify core SNPs (Single Nucleotide Polymorphism)](#ksnp) ([**kSNP3**](https://sourceforge.net/projects/ksnp/files/))
3. [Control phylogenetic tree] ([**IQ-tree2**](#iqtree) (http://www.iqtree.org))
4. [Identify high-quality SNPs using [FDA CFSAN SNP](https://snp-pipeline.readthedocs.io/en/latest/readme.html) pipeline](#snp) (in **[GalaxyTrakr](https://galaxytrakr.org/root/login?redirect=%2F)**)
5. [Annonate plasmids (long-read assembly and hybrid assembly only)](#plasmid) (**TBD**)
6. [Detect antimicrobial resistance (AMR) genes](#abricate) ([**ABRicate**](https://github.com/tseemann/abricate), [**Megares**](https://megares.meglab.org)(An antimicrobial database))
7. [Whole genome annotation and pangenome analysis](#prokka) ([**Prokka**](https://github.com/tseemann/prokka), [**Roary**](https://github.com/sanger-pathogens/Roary), [**Scoary**](https://github.com/AdmiralenOla/Scoary))

<a name = "btyper"></a>
## Detecct genes of interest of Bacillus cereus ##

<a name = "ksnp"></a>
## Identify core SNPs (Single Nucleotide Polymorphism) ##

<a name = "iqtree"></a>
## Control phylogenetic tree ##

<a name = "snp"></a>
## Identify high-quality SNPs using FDA CFSAN SNP pipeline ##

<a name = "plasmid"></a>
## Annonate plasmids (long-read assembly and hybrid assembly only) ##

<a name = "abricate"></a>
## Detect antimicrobial resistance (AMR) genes ##

<a name = "prokka"></a>
## Whole genome annotation and pangenome analysis ##
