# Short read assembly using SPAdes

## Table of Contents ##

1. [Performs basic quality control](#fastqc) [(**fastqc**)](https://github.com/s-andrews/FastQC)
2. [Trim the adapters and poor-quality bases](#trimmomatic) [(**trimmomatic**)](https://github.com/usadellab/Trimmomatic)
3. [Assemble reads into contigs](#spades) [(**SPAdes**)](https://github.com/ablab/spades)
4. [Genomic assemblies evaluation and comparison](#quast) [(**QUAST**)](https://github.com/ablab/quast)
5. [Calculating average coverage of the genome](#average_coverage) ([**BWA**](https://github.com/lh3/bwa) and [**Samtools**](https://github.com/samtools/samtools))
6. [Automation of the process](#automation)

<a name = "fastqc"></a>
### Performs basic quality control ###

<a name = "trimmomatic"></a>
### Trim the adapters and poor-quality bases ###

<a name = "spades"></a>
### Assemble reads into contigs ###

<a name = "quast"></a>
### Genomic assemblies evaluation and comparison ###

<a name = "average_coverage"></a>
### Calculating average coverage of the genome ###

<a name = "automation"></a>
## Automation of the process ##
